### ISSUE TITLE
Active response IP was not added to `iptable`

### ISSUE THREAD

**Issue by fcaffieri** [wazuh/wazuh-qa#4336]
* **Do you want to request a *feature* or report a *bug*?**
  - [X] bug report
  - [ ] feature request

* **What is the current behavior?**

The following error is produced in the `Test_integration` pipeline for Active response module:

```
08:46:30                  if not flag and command_id == 0:
08:46:30  >                   raise AssertionError("IP was not added to iptable")
08:46:30  E                   AssertionError: IP was not added to iptable
08:46:30  
08:46:30  test_active_response/test_execd/test_execd_firewall_drop.py:309: AssertionError
```

Full output:
[centos_agent_tests_Test_integration_B41373_20230720122956.txt](https://github.com/wazuh/wazuh-qa/files/12114339/centos_agent_tests_Test_integration_B41373_20230720122956.txt)

Testing the 4.6.0 Wazuh version: https://ci.wazuh.info/job/Test_integration/41373/

> Further research is required

**Comment by mauromalara** [comment]
The `4.6.0-pre-alpha1` tag will be generated again because it was generated with an old branch.

**Comment by jnasselle** [comment]
This still happens on https://github.com/wazuh/wazuh/issues/18005

**Comment by mauromalara** [comment]
On hold because a task with higher priority was assigned to me.

**Comment by mauromalara** [comment]
## Update

- I configured a CentOS 7 agent (Docker) to test in a similar environment
- I debugged the test and I found a block of code checking IP tables but failing to get the IP on time. So, it was a ""race condition"" between AR and the test trying to get the output of a command just in time:
https://github.com/wazuh/wazuh-qa/blob/6b81073733b764fd5c689e450b6a8f1e3a30dcbc/tests/integration/test_active_response/test_execd/test_execd_firewall_drop.py#L301-L311
- I applied the fix and tested it locally several times, it looks ok

**Comment by Rebits** [comment]
Removing issue from release testing objective due to regarding @mauromalara [analysis](https://github.com/wazuh/wazuh-qa/issues/4336#issuecomment-1648800226) it seems an error in the IT and not in the product

**Comment by mauromalara** [comment]
## Update

- Applied requested changes in PR

**Comment by mauromalara** [comment]
## Update 01/08/23

- Changed the way the callbacks were searching logs
- Changed the way the logs are checked by FileMonitor in the test
- Added a validation to check the AR command

**Comment by mauromalara** [comment]
## Update

- Changed how the internal options are configured
- Moved callback to framework library
- Updated docs
- Added more checks

**Comment by mauromalara** [comment]
Reopen because of an error while running the tests locally and in Jenkins (not tested in Windows).

**Comment by mauromalara** [comment]
Close because it will be fixed in this issue #4512
